build.zig

```diff
@@ -22,23 +21,114 @@ pub fn build(b: *std.Build) !void {
         "example/sysjs_generated.js",
     );
 
-    const wasm32_freestanding = std.zig.CrossTarget{ .cpu_arch = .wasm32, .os_tag = .freestanding };
-    const lib = b.addSharedLibrary(.{
+    const app = try App.init(b, .{
         .name = "example",
-        .root_source_file = .{ .path = "example/main.zig" },
+        .src = "example/main.zig",
         .optimize = optimize,
-        .target = wasm32_freestanding,
+        .deps = &[_]App.Dependency{App.Dependency{
+            .name = "sysjs",
+            .binding_file = "example/sysjs_generated.js",
+            .wrapper_file = "example/sysjs_generated.zig",
+        }},
     });
-    lib.rdynamic = true;
 
-    const example_install = b.addInstallArtifact(lib);
     var example_compile_step = b.step("example", "Compile example");
-    example_compile_step.dependOn(&example_install.step);
+    example_compile_step.dependOn(&app.install.step);
 
-    const test_step = b.step("test", "Run library tests");
-    test_step.dependOn(&testStep(b, optimize, target).step);
+    // const test_step = b.step("test", "Run library tests");
+    // test_step.dependOn(&testStep(b, optimize, target).step);
```

Comment:
Maybe just remove `testStep` and these lines entirely for now; `testStep` function is an anti-pattern I've been trying to eliminate from our codebases (prefer just writing that code directly in `pub fn build`)